### ISSUE TITLE
PLC1802 false negative when a sequence is an attribute

### ISSUE THREAD

**Issue by leandrobbraga** [astral-sh/ruff#22780]
PLC1802 won’t trigger if the sequence is an attribute.

https://play.ruff.rs/f9f8f76b-c468-4f4f-a3b8-0bca5850daf1

# Example

By running
```console
$ echo 'class Fruits:
      fruits: list[str]

      def __init__(self, fruits: list[str]):
          self.fruits = fruits

  fruits_class = Fruits(["orange", "apple"])
  fruits_list = ["a", "b"]

  if len(fruits_class.fruits):
      ...

  if len(fruits_list):
      ...
  ' | cargo run -p ruff -- check --select PLC1802 -
```

We get this output:

```console
PLC1802 [*] `len(fruits_list)` used as condition without comparison
  --> -:13:4
   |
11 |     ...
12 |
13 | if len(fruits_list):
   |    ^^^^^^^^^^^^^^^^
14 |     ...
   |
help: Remove `len`

Found 1 error.
[*] 1 fixable with the `--fix` option.
```

**Comment by leandrobbraga** [comment]
It seems related to ruff giving up on resolving `Expr::Attribute` in `From<&Expr> for ResolvedPythonType` [here](https://github.com/astral-sh/ruff/blob/main/crates/ruff_python_semantic/src/analyze/type_inference.rs#L348-L353).

```rust
impl From<&Expr> for ResolvedPythonType {
    fn from(expr: &Expr) -> Self {
        match expr {
            // Primitives.
            Expr::Dict(_) => ResolvedPythonType::Atom(PythonType::Dict),
            Expr::DictComp(_) => ResolvedPythonType::Atom(PythonType::Dict),
            // (...)
            Expr::Lambda(_)
            | Expr::Await(_)
            | Expr::Yield(_)
            | Expr::YieldFrom(_)
            | Expr::Compare(_)
            | Expr::Call(_)
            | Expr::Attribute(_)
            | Expr::Subscript(_)
            | Expr::Starred(_)
            | Expr::Name(_)
            | Expr::Slice(_)
            | Expr::IpyEscapeCommand(_) => ResolvedPythonType::Unknown,
        }
    }
}
```

**Comment by ntBre** [comment]
I think this would probably follow a code path more like `is_indirect_sequence`

https://github.com/astral-sh/ruff/blob/9f34b4634d407b4a2fd1257bfea65f5b4c0eacf0/crates/ruff_linter/src/rules/pylint/rules/len_test.rs#L130-L133

than `is_sequence`, which uses `ResolvedPythonType` because we have to resolve the definition and other assignments to the attribute. I think it would be very tricky to do that reliably without type inference.

**Comment by leandrodamascena** [comment]
I had the same problem and was about to open a new issue, but I found this one. I just submitted a PR for review.
